sdk/python/kfp/compiler/main.py

```diff
@@ -14,16 +14,14 @@
 
 
 import argparse
+from typing import Optional
 from kfp import dsl
 import kfp.compiler
 import os
-import shutil
-import subprocess
 import sys
-import tempfile
 from deprecated.sphinx import deprecated
 
-
+KF_PIPELINES_COMPILER_MODE_ENV = 'KF_PIPELINES_COMPILER_MODE'
```

Comment:
nit: 2 empty lines between this and the following function definition.

Also maybe consider prefixing the varaible name with underscore: `_KF_PIPELINES_COMPILER_MODE_ENV = ...`? (assuming it's private, and referenced only within this file)

Reply:
updated.

I'm not sure why existing env vars did not have the prefix, but I think I agree with you it should probably be internal.